test/integration/sys-mac.int.c

```diff
@@ -0,0 +1,151 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*******************************************************************************
+ * Copyright 2017-2018, Fraunhofer SIT sponsored by Infineon Technologies AG
+ * All rights reserved.
+ *******************************************************************************/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "tss2_tpm2_types.h"
+#include "tss2_sys.h"
+#include "util/tpm2b.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include "tss2_tcti.h"
+#include "tss2_sys.h"
+
+#define LOGMODULE test
+#include "util/log.h"
+#include "test.h"
+#include "sys-util.h"
+
+int
+test_sys_mac(TSS2_SYS_CONTEXT * sys_context)
+{
+
+    TSS2_RC rc;
+    TPM2B_PUBLIC out_public = { 0 };
+    TPM2B_CREATION_DATA creation_data = {0,};
+    creation_data.size = 0;
+    TPM2B_DIGEST creation_hash = {sizeof(TPM2B_DIGEST)-2,};
```

Comment:
```C
TPM2B_DIGEST creation_hash = {
        .size = sizeof(creation_hash.buffer)
};
```
I am assuming the -2 is from dropping the size field of UINT16 on the TPM2B_DIGEST and only accounting for the buffer. This is a better way to do that without magic numbers.